src/SFA.DAS.AssessorService.Application/Handlers/Staff/StaffSearchHandler.cs

```diff
@@ -35,39 +39,56 @@ public class StaffSearchHandler : IRequestHandler<StaffSearchRequest, PaginatedL
-                return new PaginatedList<StaffSearchResult>(new List<StaffSearchResult>(), 0, request.Page, 10);
+                return new StaffSearchResult
+                {
+                    EndpointAssessorOrganisationId = String.Empty,
+                    StaffSearchItems =
+                        new PaginatedList<StaffSearchItems>(new List<StaffSearchItems>(), 0, request.Page, 10)
+                };
 
             var pageSize = 10;
             var searchResult = await Search(request);
             var totalRecordCount = searchResult.TotalCount;
 
+            var displayEpao = false;
             if (searchResult.TotalCount == 0)
             {
                 totalRecordCount = await _staffIlrRepository.CountLearnersByName(request.SearchQuery);
-               searchResult.PageOfResults = await _staffIlrRepository.SearchForLearnerByName(request.SearchQuery, request.Page, pageSize);   
+                searchResult.PageOfResults = await _staffIlrRepository.SearchForLearnerByName(request.SearchQuery, request.Page, pageSize);
+            }
+            else
+            {
+                displayEpao = searchResult.DisplayEpao;
             }
 
             _logger.LogInformation(searchResult.PageOfResults.Any() ? LoggingConstants.SearchSuccess : LoggingConstants.SearchFailure);
 
-            var searchResults = Mapper.Map<List<StaffSearchResult>>(searchResult.PageOfResults);
+            var searchResults = Mapper.Map<List<StaffSearchItems>>(searchResult.PageOfResults);
 
             searchResults = MatchUpExistingCompletedStandards(searchResults);
             searchResults = PopulateStandards(searchResults, _assessmentOrgsApiClient, _logger);
 
-            return new PaginatedList<StaffSearchResult>(searchResults, totalRecordCount, request.Page, pageSize);
+            return new StaffSearchResult
+            {
+                DisplayEpao = displayEpao,
+                EndpointAssessorOrganisationId = displayEpao && searchResults.Count > 0 ? searchResults.First().EndpointAssessorOrganisationId : string.Empty,
+                StaffSearchItems = new PaginatedList<StaffSearchItems>(searchResults, totalRecordCount, request.Page, pageSize)
+            };
         }
 
         private async Task<StaffReposSearchResult> Search(StaffSearchRequest request)
         {
             // Naive decision on what is being searched.
 
-            var regex = new Regex(@"\b(EPA|epa)[0-9]{4}\b");
+            var regex = new Regex(@"\b(?i)(epa)[0-9]{4}\b");
             if (regex.IsMatch(request.SearchQuery))
-            {
-                return await _staffIlrRepository.SearchForLearnerByEpaOrgId(request);
+            {                
+                var sr = await _staffIlrRepository.SearchForLearnerByEpaOrgId(request);
+                sr.DisplayEpao = true;
+                return sr;
             }
 
             if (request.SearchQuery.Length == 10 && long.TryParse(request.SearchQuery, out var uln))
```

Comment:
Don't know this domain, so this is just a query that can be ignored.  Is length 10 and only 10 right here?